packages/compiler/src/core/checker.ts

```diff
@@ -1753,17 +1770,31 @@ export function createChecker(program: Program): Checker {
     });
   }
 
-  function getSymbolLinks(s: Sym): SymbolLinks {
+  function getSymbolLinks(s: Sym, node?: Node): SymbolLinks {
     const id = getSymbolId(s);
 
     if (symbolLinks.has(id)) {
-      return symbolLinks.get(id)!;
+      // Check if the link has an attached deprecation
+      const links = symbolLinks.get(id)!;
```

Comment:
If we are going to have the deprecation data added to the symbol links do we then need the stateMap at all?